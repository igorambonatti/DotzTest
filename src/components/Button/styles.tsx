import { RectButton } from 'react-native-gesture-handler'

import styled from 'styled-components/native'

export const Container = styled(RectButton)`
  justify-content: center;
  align-items: center;
  height: 60px;
  background: #fff;
  border-radius: 10px;
  margin-top: 8px;
`

export const ButtonText = styled.Text`
  color: #f56700;
  font-size: 18px;
  font-weight: bold;
`
