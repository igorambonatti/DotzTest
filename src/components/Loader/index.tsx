import React from 'react'
import { View, ActivityIndicator } from 'react-native'

// import { Container } from './styles';

const Loader: React.FC = () => {
  return (
    <View
      style={{
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F56700',
      }}
    >
      <ActivityIndicator size='large' color='#fff' />
    </View>
  )
}

export default Loader
