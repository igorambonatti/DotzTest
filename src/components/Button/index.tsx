import React from 'react'
import { RectButtonProperties } from 'react-native-gesture-handler'

import { Container, ButtonText } from './styles'

export interface ButtonProps extends RectButtonProperties {
  children: string
}

const Button: React.FC<ButtonProps> = ({ children, ...rest }) => (
  <Container testID={'button'} {...rest}>
    <ButtonText>{children}</ButtonText>
  </Container>
)

export default Button
