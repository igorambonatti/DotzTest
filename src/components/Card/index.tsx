import React from 'react'
import { View, Text } from 'react-native'
import { ViewProps } from 'react-native'
import FontAwesome from 'react-native-vector-icons/FontAwesome'

import * as S from './styles'

export interface CardProps extends ViewProps {
  icon: string
  title: string
  onSelect: () => void
}

const Card: React.FC<CardProps> = ({ icon, title, onSelect, ...rest }) => {
  return (
    <S.ButtonSelect onPress={onSelect} testID={'button'}>
      <S.Container {...rest}>
        <S.IconView>
          <FontAwesome name={icon} size={24} color='#fff' />
        </S.IconView>
        <S.Title testID={'title'}>{title}</S.Title>
      </S.Container>
    </S.ButtonSelect>
  )
}

export default Card
