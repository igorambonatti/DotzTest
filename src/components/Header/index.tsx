import React from 'react'
import FontAwesome from 'react-native-vector-icons/FontAwesome'

import { Body, Header as HeaderNB, Left, Right } from 'native-base'

import * as S from './styles'

export interface HeaderProps {
  title?: string
  onBack?: () => void
}

const Header: React.FC<HeaderProps> = ({ onBack, title, ...rest }) => {
  return (
    <S.Container
      androidStatusBarColor='#FFF'
      iosBarStyle='dark-content'
      noShadow
      {...rest}
      testID={'container'}
    >
      <Left>
        <S.BackButton onPress={onBack}>
          <FontAwesome name='angle-left' size={36} color='#fff' />
        </S.BackButton>
      </Left>
      <Body
        style={{
          flex: Platform.OS !== 'ios' ? 6 : 1,
          alignItems: 'center',
          left: Platform.OS !== 'ios' ? 20 : 0,
        }}
      >
        <S.Title>{title}</S.Title>
      </Body>
      <Right />
    </S.Container>
  )
}

export default Header
