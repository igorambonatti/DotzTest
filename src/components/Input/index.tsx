import React, {
  useEffect,
  useCallback,
  useRef,
  useImperativeHandle,
  forwardRef,
  useState,
} from 'react'
import { TextInputProps, Text } from 'react-native'

import { useField } from '@unform/core'

import * as S from './styles'

export interface InputProps extends TextInputProps {
  name: string
}

interface InputValueReference {
  value?: string
}

interface InputRef {
  focus(): void
}

const Input: React.FC<InputRef, InputProps> = (
  { name, icon, containerStyle = {}, ...rest },
  ref,
) => {
  const inputElementRef = useRef<any>(null)

  const { registerField, defaultValue = '', fieldName, error } = useField(name)
  const inputValueRef = useRef<InputValueReference>({ value: defaultValue })

  const [isFocused, setIsFocused] = useState(false)
  const [isFilled, setIsFilled] = useState(false)

  const handleInputFocus = useCallback(() => {
    setIsFocused(true)
  }, [])

  const handleInputBlur = useCallback(() => {
    setIsFocused(false)
    setIsFilled(!!inputValueRef.current.value)
  }, [])

  useImperativeHandle(ref, () => ({
    focus() {
      inputElementRef.current.focus()
    },
  }))

  useEffect(() => {
    registerField<string>({
      name: fieldName,
      ref: inputValueRef.current,
      path: 'value',
    })
  }, [fieldName, registerField])

  return (
    <>
      {error && <S.LabelError>{error}</S.LabelError>}
      <S.Container
        style={containerStyle}
        isFocused={isFocused}
        isErrored={!!error}
      >
        <S.TextInput
          testID={'input'}
          ref={inputElementRef}
          keyboardAppearance='dark'
          placeholderTextColor='#fff'
          defaultValue={defaultValue}
          onFocus={handleInputFocus}
          onBlur={handleInputBlur}
          onChangeText={(value) => {
            inputValueRef.current.value = value
          }}
          {...rest}
        />
      </S.Container>
    </>
  )
}

export default forwardRef(Input)
