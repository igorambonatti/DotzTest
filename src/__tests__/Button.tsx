import React from 'react'
import { create, act } from 'react-test-renderer'

import { ButtonProps } from '../components/Button'
import Button from '../components/Button'

let props: ButtonProps = {
  onPress: jest.fn(),
  children: 'Submit',
}

const tree = create(<Button {...props} />)

test('snapshot', () => {
  expect(tree).toMatchSnapshot()
})
it(`render component`, async () => {
  const button = tree.root.findByProps({ testID: 'button' }).props
  expect(button).toBeTruthy()
})

test('button press', () => {
  const button = tree.root.findByProps({ testID: 'button' }).props
  act(() => button.onPress())
})
