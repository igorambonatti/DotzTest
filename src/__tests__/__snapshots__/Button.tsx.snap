// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RNGestureHandlerButton
  activeOpacity={0.105}
  collapsable={false}
  onActiveStateChange={[Function]}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  onPress={[MockFunction]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "height": 60,
      "justifyContent": "center",
      "marginTop": 8,
    }
  }
  testID="button"
  underlayColor="black"
>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "borderBottomLeftRadius": undefined,
        "borderBottomRightRadius": undefined,
        "borderRadius": 10,
        "borderTopLeftRadius": undefined,
        "borderTopRightRadius": undefined,
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#f56700",
          "fontSize": 18,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Submit
  </Text>
</RNGestureHandlerButton>
`;
